pandas/tests/extension/base/groupby.py

```diff
@@ -56,6 +56,39 @@ def test_groupby_agg_extension(self, data_for_grouping):
         result = df.groupby("A").first()
         self.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "expected",
+        [
+            pd.DataFrame(
+                {
+                    "td": {
+                        0: pd.Timedelta("0 days 01:00:00"),
+                        1: pd.Timedelta("0 days 01:15:00"),
+                        2: pd.Timedelta("0 days 01:15:00"),
+                    }
+                }
+            )
+        ],
+    )
+    def test_groupby_agg_extension_timedelta_cumsum_with_named_aggregation(
+        self, expected
+    ):
+        ts = pd.DatetimeIndex(
+            [
+                pd.Timestamp("2021/01/01 00:30"),
+                pd.Timestamp("2021/01/01 00:45"),
+                pd.Timestamp("2021/01/01 02:00"),
+            ]
+        )
+        df = pd.DataFrame({"value": [1, 2, 3], "ts": ts})
+        df["td"] = df["ts"] - df["ts"].shift(1, fill_value=ts[0] - pd.Timedelta("1h"))
+        df["amount"] = df["value"] * 10
+        df.loc[:2, "grps"] = "a"
+        df.loc[2:, "grps"] = "b"
+        agg_rules = {"td": ("td", "cumsum")}
+        result = df.groupby("grps").agg(**agg_rules)
```

Comment:
can you combine these two as `.agg(td=("td", "cumsum"))`

Reply:
Also, this is a bit of a nit pick, but it is best to separate out the call to `groupby` from the call to `agg`. These makes setting breakpoints easier for when developers have to debug later on.

```
gb = df.groupby("grps")
result = gb.agg(td=("td", "cumsum"))
```

Reply:
Sure, this seems reasonable to me! 
I have updated the code here: d12629013a9e68981b7715bd0f53d8851f8bdb0f